p2pspec

```diff
@@ -1,4 +1,36 @@
-:::NEW p2p spec:::
+:::POTATO MESSAGES:::
+
+:establishing connection:
+- A->B connect request [conn_uuid, challenge, identity {A, B}]
+  - spam when initiating connection
+  - retry every 10 minutes
+- B->A connect response [conn_uuid, B signed challenge, challenge]
+  - B now stores {conn_uuid, A} in active connections
+- A->B connect response response [conn_uuid, A signed challenge]
+  - A now stores {conn_uuid, B} in active connections
+- A<->B ping
+  - periodic message to indicate the connection is still alive
+
+:blocks:
+- A->B request block
+  - B server caches request {A, blockrange} and starts serving
+  - A should spam this request to ensure receipt
+    - can stop spamming when first block is received
```

Comment:
how about request block range? Request message has [my_head, my_head - 10, unknown_hash]
It is responded by sending a range of blocks (best guess at unknown range)